nltk/tokenize/stanford_segmenter.py

```diff
@@ -64,13 +65,20 @@ def __init__(self,
                 env_vars=('STANFORD_SEGMENTER',),
                 searchpath=(), url=_stanford_url,
                 verbose=verbose)
-        slf4j = find_jar(
+        if path_to_slf4j is not None:
+            if path_to_slf4j == 'PLACEHOLDER':
+                path_to_slf4j = None
+            slf4j = find_jar(
                 self._SLF4J, path_to_slf4j,
                 env_vars=('SLF4J', 'STANFORD_SEGMENTER',),
                 searchpath=(), url=_stanford_url,
                 verbose=verbose)
+        else:
+            slf4j = None
 
-        # This is passed to java as the -cp option, the segmenter needs slf4j.
+        # This is passed to java as the -cp option, the old version of segmenter needs slf4j.
+        # The new version of stanford-segmenter-2016-10-31 doesn't need slf4j
```

Comment:
OK. I have deleted _SLF4J. 
As I have recorded, as long as the second parameter transmitted to find_jar is not None, the find_jar_iter would execute without environment parameters. So If the user only input 'slf4j-api.jar' instead of the path to slf4j-api.jar, I need to set this parameter to None.